### ISSUE TITLE
Native logging driver's fluentd DaemonSet consumes all available memory

### ISSUE THREAD

**Issue by marcusianlevine** [rancher/rancher#16329]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1. Deploy Rancher on small nodes with native logging driver enabled (in our case sending to Elasticsearch)
1. Generate a large volume of logs

**Result:**
Ever-increasing memory footprint of `fluentd` DaemonSet

**Other details that may be helpful:**
Adding default resource constraints to the `fluentd` DaemonSet (or better yet customizable constraints) would solve this issue.

We'll simply modify the DaemonSet after deployment for now, but Rancher cluster services should not suck up all available memory by default.

**Environment information**
- Rancher version: 2.1.1
- Installation option (single install/HA): HA (Helm)

- Docker (use `docker info`): 17.09
```
Server Version: 17.09.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 451
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-1065-aws
Operating System: Ubuntu 16.04.5 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.22GiB
```

**Comment by marcusianlevine** [comment]
Doing some more research on `fluentd` memory consumption, it looks like the root cause of the runaway memory consumption may be this [recently fixed memory leak associated with tailing rotating logs](https://github.com/fluent/fluentd/issues/1941#issuecomment-412689268)

If that's the case, simply upgrading to `fluentd==1.2.5` might resolve the specific issue we're facing, but I still think all Rancher internal services should have sane resource constraints (or at least the ability to set them)

**Comment by deniseschannon** [comment]
@aiwantaozi @loganhz what version of fluentd are we using for logging?

**Comment by aiwantaozi** [comment]
Rancher 2.1.1 deploy fluentd v1.1, which have the memory leak issue, we upgrade fluentd to v1.3.3 in Rancher 2.2

**Comment by deniseschannon** [comment]
This should be fixed with Rancher 2.2.

**Comment by qeternity** [comment]
This is definitely not fixed
